app/src/editor/canvas/components/ModuleSearch.jsx

```diff
@@ -24,31 +59,69 @@ export default class ModuleSearch extends React.PureComponent {
 
     async load() {
         const modules = await getModuleTree()
+        this.allModules = modules
         if (this.unmounted) { return }
-        this.setState({ modules })
+        this.setState({
+            // Default to showing all modules
+            matchingModules: moduleTreeSearch(this.allModules, ''),
+        })
     }
 
-    onChange = (event) => {
+    onChange = (event: any) => {
         const { value } = event.currentTarget
-        this.setState({ search: value })
+
+        // Search modules
+        const matchingModules = moduleTreeSearch(this.allModules, value)
+        this.setState({
+            search: value,
+            matchingModules,
+        })
+
+        // Search streams
+        this.props.updateFilter({
+            id: '',
+            search: value,
+            sortBy: 'lastUpdated',
+            order: 'desc',
+            uiChannel: false,
+            public: true,
+        })
```

Comment:
So this updates the "global" stream filter… affecting every piece of UI that uses the stream filter. That doesn't seem right to me. 

e.g. If you search for a module, this search becomes the value of the search box in the userpages streams UI

![weird-global-filter](https://user-images.githubusercontent.com/43438/54183716-c5e49b00-44df-11e9-9013-e4b0335eab61.gif)


Disclosure: This doesn't *actually* occur currently because the link to the streams in the top nav currently happen to be regular `<a href` anchors, thus they triggering a page reload, I did have to modify the links to be `<Link>`s to get this behaviour but those should probably be `<Link`s anyway so I did cheat but I think it highlights the issue.

IMO overall simpler if this component manages its own state and streams data.